### PR TITLE
Making CreateCodeWriter protected so that it can be overridden in Mvc tests.

### PR DESCRIPTION
**Pull request by sornaks**

### CODE REVIEW

File: src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -24,8 +24,8 @@ public CSharpCodeGenerator(CodeGeneratorContext context)
         private ChunkTree Tree { get { return Context.ChunkTreeBuilder.ChunkTree; } }
         public RazorEngineHost Host { get { return Context.Host; } }
 
-        // Internal for testing
-        internal virtual CSharpCodeWriter CreateCodeWriter()
+        // Protected for testing
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeGenerator.cs]: Put this in a `<summary>` in-case someone tries to override it.

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by sornaks** [comment]
Thanks! https://github.com/aspnet/Razor/commit/1f480386f4fd36aacb4e0f3959469901aab67ed9

**Pull request closed without merging**